### PR TITLE
ValuesSource refactoring: Wire up string_stats aggregation

### PR DESCRIPTION
**Pull request by csoulios**
Wire up string_stats aggregation for #42949

### CODE REVIEW

File: x-pack/plugin/analytics/src/test/java/org/elasticsearch/xpack/analytics/stringstats/StringStatsAggregatorTests.java [hunk 1]
```diff
@@ -259,4 +270,22 @@ public void testNestedAggregation() throws IOException {
...
 
+
+//    @Override
+//    protected Collection<Class<? extends Plugin>> getPlugins() {
+////        List<Class<? extends Plugin>> plugins = new ArrayList<>(super.getPlugins());
+////        plugins.add(AggregateMetricMapperPlugin.class);
+////        plugins.add(LocalStateCompositeXPackPlugin.class);
+////        return plugins;
+//        return null;
+//    }
```
Comment (not-napoleon) [inline x-pack/plugin/analytics/src/test/java/org/elasticsearch/xpack/analytics/stringstats/StringStatsAggregatorTests.java]: Do we need to keep this?

**Review by not-napoleon — APPROVED** [review]
LGTM

**Comment by csoulios** [comment]
@elasticmachine run elasticsearch-ci/bwc

File: x-pack/plugin/analytics/src/test/java/org/elasticsearch/xpack/analytics/stringstats/StringStatsAggregatorTests.java (hunk 1 above)
Comment (csoulios) [reply x-pack/plugin/analytics/src/test/java/org/elasticsearch/xpack/analytics/stringstats/StringStatsAggregatorTests.java]: Sorry, I missed cleaning this up. Fixed

**Pull request merged**